examples/mxnet/gat/gat_batch.py

```diff
@@ -168,39 +187,55 @@ def main(args):
                 args.attn_drop,
                 args.residual)
 
-    model.initialize(ctx=ctx)
+    model.initialize()
+
+    if cuda:
+        model.collect_params().reset_ctx(ctx)
```

Comment:
Why not just use the original way `model.initialize(ctx=ctx)`, which looks much cleaner